types/hapi-auth-jwt2/index.d.ts

```diff
@@ -1,39 +1,43 @@
-// Type definitions for hapi-auth-jwt2 7.0
+// Type definitions for hapi-auth-jwt2 8.0
 // Project: https://github.com/dwyl/hapi-auth-jwt2
 // Definitions by: Warren Seymour <https://github.com/warrenseymour>
+//                 Simon Schick <https://github.com/SimonSchick>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.4
 
-import { Request, Response, PluginFunction } from 'hapi';
+import { Request, ResponseObject, Plugin, ResponseToolkit } from 'hapi';
+import { VerifyOptions } from 'jsonwebtoken';
 
-declare namespace hapiAuthJwt2 {
-    /**
-     * A key lookup function
-     *
-     * @param decoded the *decoded* but *unverified* JWT received from client
-     * @param callback the key lookup callback
-     */
-    type KeyLookup = (decoded: any, callback: KeyLookupCallback) => void;
+declare module 'hapi' {
+    interface ServerAuth {
+        strategy(name: string, scheme: 'jwt', options?: hapiAuthJwt2.Options): void;
+    }
+}
 
-    /**
-     * Called when key lookup function has completed
-     *
-     * @param err an internal error
-     * @param key the secret key
-     * @param extraInfo any additional information that you would like
-     * to use in `validateFunc` which can be accessed via
-     * `request.plugins['hapi-auth-jwt2'].extraInfo`
-     */
-    type KeyLookupCallback = (err: any, key: string, extraInfo?: any) => void;
+declare namespace hapiAuthJwt2 {
+    interface ExtraInfo {
```

Comment:
Is generally a bad idea to have empty interface. Remove this.

Reply:
See comment below.